src/WaveFunction/WaveFunction.cpp

```diff
@@ -62,4 +63,188 @@ double WaveFunction::calculateShannonEntropy() const {
+        for (size_t I_alpha = 0; I_alpha < dim_alpha; I_alpha++) {
+            if (!alpha.isOccupied(m)) {
+                for (size_t e1 = 0; e1 < N_alpha; e1++) {  // e1 (electron 1) loops over the (number of) electrons
+                    size_t p = alpha.get_occupation_index(e1);  // retrieve the index of a given electron
+
+                    if (p < m) {
+                        size_t address = I_alpha - fock_space_alpha.get_vertex_weights(p, e1 + 1);
+                        size_t e2 = e1 + 1;
+                        size_t q = p + 1;
+                        int sign = 1;
+
+                        fock_space_alpha.shiftUntilNextUnoccupiedOrbital<1>(alpha, address, q, e2, sign);
+                        while (q != m) {
+                            q++;
+                            fock_space_alpha.shiftUntilNextUnoccupiedOrbital<1>(alpha, address, q, e2, sign);
+                        }
+
+                        address += fock_space_alpha.get_vertex_weights(q, e2);
+
+                        for (size_t I_beta = 0; I_beta < dim_beta; I_beta++) {
+                            correction_coefficients(I_alpha * dim_beta + I_beta) +=
+                                    sign * t(p, m) * current_coefficients(address * dim_beta + I_beta);
+                        }
+                    }
+
+                    if (p > m) {
+                        size_t address = I_alpha - fock_space_alpha.get_vertex_weights(p, e1 + 1);
+                        size_t e2 = e1 - 1;
+                        size_t q = p - 1;
+                        int sign = 1;
+
+                        fock_space_alpha.shiftUntilPreviousUnoccupiedOrbital<1>(alpha, address, q, e2, sign);
+                        while (q != m) {
+                            q--;
+                            fock_space_alpha.shiftUntilPreviousUnoccupiedOrbital<1>(alpha, address, q, e2, sign);
+                        }
+
+                        address += fock_space_alpha.get_vertex_weights(q, e2 + 2);
+                        for (size_t I_beta = 0; I_beta < dim_beta; I_beta++) {
+                            correction_coefficients(I_alpha * dim_beta + I_beta) +=
+                                    sign * t(p, m) * current_coefficients(address * dim_beta + I_beta);
+                        }
+                    }
+                }
+
+
+            } else {  // If orbital m is occupied we can perform an in-place operation
+                for (size_t I_beta = 0; I_beta < dim_beta; I_beta++) {
+                    correction_coefficients(I_alpha * dim_beta + I_beta) +=
+                            (t(m, m) - 1) * current_coefficients(I_alpha * dim_beta + I_beta);
+                }
+            }
+
+
+            if (I_alpha < dim_alpha - 1) {  // prevent the last permutation to occur
+                fock_space_alpha.setNextONV(alpha);
+            }
+        }
+
+        current_coefficients += correction_coefficients;
+        correction_coefficients.setZero();
```

Comment:
I'm not sure, but is it possible to add the total correction at the end instead of resetting?

Reply:
No, the corrected `current_coefficients` are required in each iteration.